StandardLibrary/runtimes/python/src/standard_library/internaldafny/extern/DafnyLibraries.py

```diff
@@ -0,0 +1,126 @@
+import _dafny
+from pathlib import Path
+import threading
+
+import standard_library.internaldafny.generated.DafnyLibraries
+from standard_library.internaldafny.generated.DafnyLibraries import *
+import standard_library.internaldafny.generated.Wrappers as Wrappers
+
+# This is copy-pasted from DafnyStandardLibraries:
+# https://github.com/dafny-lang/dafny/blob/f01af4a4e86a038ed4ea9f81464b2c9bca1955e4/Source/DafnyStandardLibraries/src/Std_Concurrent.py
+
+class Lock:
+    def ctor__(self):
+        pass
+        
+    def __init__(self) -> None:
+        self.lock = threading.Lock()
+
+    def Lock__(self):
+        self.lock.acquire()
+
+    def Unlock(self):
+        self.lock.release()
+
+
+class MutableMap(standard_library.internaldafny.generated.DafnyLibraries.MutableMap):
+    def ctor__(self):
+        pass
+        
+    def __init__(self) -> None:
+        self.map = dict()
+        self.lock = Lock()
+
+    def Keys(self):
+        self.lock.Lock__()
+        s = self.map.keys()
+        self.lock.Unlock()
+        return _dafny.Set(s)
+
+    def HasKey(self, k):
+        self.lock.Lock__()
+        b = k in self.map
+        self.lock.Unlock()
+        return b
+
+    def Values(self):
+        self.lock.Lock__()
+        s = self.map.values()
+        self.lock.Unlock()
+        return _dafny.Set(s)
+
+    def Items(self):
+        self.lock.Lock__()
+        s = self.map.items()
+        self.lock.Unlock()
+        return _dafny.Set(s)
+
+    def Put(self, k, v):
+        self.lock.Lock__()
+        self.map[k] = v
+        self.lock.Unlock()
+
+    def Get(self, k):
+        self.lock.Lock__()
+        try:
+            v = self.map.get(k)
+        except KeyError:
+            self.lock.Unlock()
+            return Wrappers.Option_None()
+        self.lock.Unlock()
+        return Wrappers.Option_Some(v)
+
+    def Remove(self, k):
+        self.lock.Lock__()
+        self.map.pop(k, None)
+        self.lock.Unlock()
+
+    def Size(self):
+        self.lock.Lock__()
+        l = len(self.map)
+        self.lock.Unlock()
+        return l
+
+# This is copy-pasted from DafnyStandardLibraries:
+# https://github.com/dafny-lang/dafny/blob/f01af4a4e86a038ed4ea9f81464b2c9bca1955e4/Source/DafnyStandardLibraries/src/Std_FileIOInternalExterns.py
+
+import traceback
+import os.path
+import pathlib
+
+class FileIO:
+    @staticmethod
+    def INTERNAL__WriteBytesToFile(path, contents):
+        path_str = path.VerbatimString(False)
+        contents_bytes = bytes(contents)
+
+        try:
+            pathlib.Path(path_str).parent.mkdir(parents=True, exist_ok=True)
+
+            with open(path_str, mode="wb") as file:
+                contents = file.write(contents_bytes)
+                return (False, _dafny.Seq())
+        except:
+            exc_str = traceback.format_exc()
+            exc_seq = _dafny.Seq(exc_str)
+            return (True, exc_seq)
```

Comment:
I believe this will overwrite a file if it exists, right? do we want that?

Reply:
Correct -- this matches behavior in Java and NET.

Reply:
Java:
* Code: https://github.com/aws/aws-cryptographic-material-providers-library/blob/main/StandardLibrary/runtimes/java/src/main/java/DafnyLibraries/FileIO.java
* Docs:https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#write-java.nio.file.Path-byte:A-java.nio.file.OpenOption...-
  * "By default the method creates a new file or overwrites an existing file"

NET:
* Code: https://github.com/dafny-lang/libraries/blob/e9b898d0ec08e129b8e61306b2617b36c4ce5d3e/src/FileIO/FileIO.cs#L79
* Docs: https://learn.microsoft.com/en-us/dotnet/api/system.io.file.writeallbytes?view=net-8.0
  * "If the target file already exists, it is truncated and overwritten."

Reply:
cool cool